tests/python/test_policy_service.py

```diff
@@ -0,0 +1,192 @@
+
+
+@pytest.fixture
+def mock_database():
+    with patch("policyengine_api.services.policy_service.database") as mock_db:
+        yield mock_db
+
+
+@pytest.fixture
+def sample_policy_data():
+    return {
+        "id": 1,
+        "country_id": "US",
+        "policy_json": json.dumps({"param": "value"}),
+        "policy_hash": "hash123",
+        "label": "test_policy",
+        "api_version": "1.0.0",
+    }
+
+
+@pytest.fixture
+def policy_service():
+    return PolicyService()
+
+
+class TestPolicyService:
+
+    test_policy_id = 8  # Arbitrary value higher than 5
+
+    def test_get_policy_success(
+        self, policy_service, mock_database, sample_policy_data
+    ):
+        # Setup mock
+        mock_database.query.return_value.fetchone.return_value = (
+            sample_policy_data
+        )
+
+        # Test
+        result = policy_service.get_policy("us", self.test_policy_id)
+
+        print(result)
+        print(result["policy_json"])
+        print(type(result["policy_json"]))
+        # Verify
+        assert result is not None
+        assert isinstance(result["policy_json"], dict)
+        assert result["policy_json"]["param"] == "value"
+        mock_database.query.assert_called_once_with(
+            "SELECT * FROM policy WHERE country_id = ? AND id = ?",
+            ("us", self.test_policy_id),
+        )
+
+    def test_get_policy_not_found(self, policy_service, mock_database):
+        # Setup mock
+        mock_database.query.return_value.fetchone.return_value = None
+
+        # Test
+        garbage_id = 999
+        result = policy_service.get_policy("us", garbage_id)
+
+        # Verify
+        assert result is None
+        mock_database.query.assert_called_once()
+
+    def test_get_policy_json(
+        self, policy_service, mock_database, sample_policy_data
+    ):
+        # Setup mock
+        mock_database.query.return_value.fetchone.return_value = {
+            "policy_json": sample_policy_data["policy_json"]
+        }
+
+        # Test
+        result = policy_service.get_policy_json("us", self.test_policy_id)
+
+        # Verify
+        assert result == sample_policy_data["policy_json"]
+        mock_database.query.assert_called_once()
+
+    def test_set_policy_new(self, policy_service, mock_database):
+        new_policy_id = 10
+
+        # Setup mocks
+        mock_database.query.return_value.fetchone.side_effect = [
+            None,  # First call for existing policy check
+            {"id": new_policy_id},  # Second call to get new policy
+        ]
+
+        test_policy = {"param": "value"}
+
+        # Test
+        policy_id, message, exists = policy_service.set_policy(
+            "US", "new_policy", test_policy
+        )
+
+        # Verify
+        assert policy_id == new_policy_id
+        assert message == "Policy created"
+        assert exists is False
+        assert (
+            mock_database.query.call_count == 3
```

Comment:
**suggestion, non-blocking** I would personally just check the calls directly instead of count.  (i.e. verify it made each query against the expected string). It doesn't involve much more work and it can catch issues (I.e. if I inject the wrong value into the query, etc.)